### DISCUSSION TITLE
What is `~/.config/yazi/init.lua`? How it's related to plugins?

### DISCUSSION THREAD

**Discussion by LinArcX** [sxyazi/yazi#595] [1. Q&A] [answered]
As I understood from [here](https://yazi-rs.github.io/docs/plugins/overview) we should create: `plugin.yazi` directory with `init.lua` file inside: `~/.config/yazi/plugins` for each plugin.

But still, I don't know how to load these plugins automatically.

For example in here: https://yazi-rs.github.io/docs/tips

you said that we can change `init.lua`. But my question is that for example can I group all the changes that are related to Status and put them here: `~/.config/yazi/plugins/status.yazi`?

P.S: it won't work.they only work in `~/.config/yazi/init.lua`

**Comment by sxyazi** [comment, marked as answer]
Just `require("status")`, see https://yazi-rs.github.io/docs/plugins/overview#sync-vs-async. Maybe we can make it clearer?

**Reply by lpnh** [reply]
In their documentation, NvChad has a [Learn basic Lua](https://nvchad.com/docs/quickstart/learn-lua) section that provides an overview of Lua. Perhaps including something similar could help users with their own custom configurations?

**Reply by sxyazi** [reply]
I have linked to https://learnxinyminutes.com/docs/lua/ in the [debugging section](https://yazi-rs.github.io/docs/plugins/overview#debugging) and provided some editor plugin suggestions, I think that would be enough :)

**Reply by lpnh** [reply]
It's perfect. It seems I myself have ended up giving a fine example of how we struggle with reading the documentation.

**Reply by LinArcX** [reply]
Thanks, `require("status")` works but interestingly `require("status"){}` will give me lots of errors.
